gpMgmt/bin/gpinitsystem

```diff
@@ -620,7 +620,10 @@ END_PYTHON_CODE
 	done
 	. $CLUSTER_CONFIG
 	NUM_DATADIR=${#DATA_DIRECTORY[@]}
-	if [ `$ECHO ${M_HOST_ARRAY[@]}|$TR ' ' '\n'|$AWK -F"~" '{print $2}'|$SORT -u|$WC -l` -ne $MCOUNT ];then
+  COUNT_MHOST_NODES=`$ECHO ${M_HOST_ARRAY[@]}|$TR ' ' '\n'|$AWK -F"~" '{print $2}'|$UNIQ -c|$AWK '{print $1}'|$SORT -u`
+  if [ `$ECHO ${COUNT_MHOST_NODES}|$WC -w` -ne 1 ];then
+	  ERROR_EXIT "[FATAL]:-Uneven distribution of network interfaces by hosts" 2
```

Comment:
Probably is is worth to add more information to show what hosts are not unique.
For example:
```
LOG_MSG "[FATAL]:-Uneven distribution of network interfaces by hosts" 1
LOG_MSG "[INFO]: Number of hosts: `$ECHO ${M_HOST_ARRAY[@]}|$TR ' ' '\n'|$AWK -F"~" '{print $2}'|$WC -w`" 1
$ECHO ${M_HOST_ARRAY[@]}|$TR ' ' '\n'|$AWK -F"~" '{print $1" -> "$2}'
LOG_MSG "[INFO]: Number of unique hosts: `$ECHO ${COUNT_MHOST_NODES}|$WC -w`" 1
ERROR_EXIT "[FATAL]:-Unable to continue" 2
```
Will produce:

> 20211213:13:36:04:025358 gpinitsystem:ssmirnov-mdw:gpadmin-[FATAL]:-Uneven distribution of network interfaces by hosts
> 20211213:13:36:04:025358 gpinitsystem:ssmirnov-mdw:gpadmin-[INFO]: Number of hosts: 3
> ssmirnov-sdw1-1 -> ssmirnov-sdw1
> ssmirnov-sdw2-1 -> ssmirnov-sdw2
> ssmirnov-sdw2-2 -> ssmirnov-sdw2
> 20211213:13:36:05:025358 gpinitsystem:ssmirnov-mdw:gpadmin-[INFO]: Number of unique hosts: 2
> 20211213:13:36:05:025358 gpinitsystem:ssmirnov-mdw:gpadmin-[FATAL]:-Unable to continue Script Exiting!
>

Reply:
For me this improvement looks nice and gives precise information to fix error.